### PR TITLE
support for flush commmand

### PR DESCRIPTION
**Pull request by srirampmalladi**
Change-Id: I381a7b537784a8d2b864caa7067d1ff579571cef

### CODE REVIEW

File: src/main/java/com/spotify/folsom/client/binary/FlushAllRequest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  private final int delay;
+
+  public FlushAllRequest(final int delay, final int opaque) {
+    super("", opaque);
+    this.delay = delay;
+  }
+
+  @Override
+  public ByteBuf writeRequest(final ByteBufAllocator alloc, final ByteBuffer dst) {
+    writeBinaryHeader(dst, OpCode.FLUSH, 0, 4, 4, 0, opaque);
```
Comment (spkrka) [inline src/main/java/com/spotify/folsom/client/binary/FlushAllRequest.java:39]: the extraLength should be either 0 or 4 here I guess, depending on the value of delay

**Comment by spkrka** [comment]
Nice! I would like to see some added IntegrationTest method too, and possibly an extra test for MisbehavingServerTest (should hopefully be very simple to add)

File: src/test/java/com/spotify/folsom/client/binary/FlushAllRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2014-2015 Spotify AB
```
Comment (spkrka) [inline src/test/java/com/spotify/folsom/client/binary/FlushAllRequestTest.java]: this should only say 2015

File: src/main/java/com/spotify/folsom/client/ascii/FlushAllRequest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    return toBuffer(alloc, dst);
+  }
+
+  @Override
+  protected void handle(AsciiResponse response) throws IOException {
+    AsciiResponse.Type type = response.type;
+    if (type == AsciiResponse.Type.OK) {
+      succeed(null);
+    } else {
+      throw new IOException("Unexpected line: " + type);
```
Comment (spkrka) [inline src/main/java/com/spotify/folsom/client/ascii/FlushAllRequest.java:50]: this line could be tested with MisbehavingServerTest

File: src/main/java/com/spotify/folsom/client/binary/FlushAllRequest.java (hunk 1 above)
Comment (srirampmalladi) [inline src/main/java/com/spotify/folsom/client/binary/FlushAllRequest.java:39]: Actually, I will remove the if condition in the next line so we write the delay whatever its value is.

Comment (spkrka) [inline src/main/java/com/spotify/folsom/client/binary/FlushAllRequest.java:39]: Sounds good!

**Pull request closed without merging**